src/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,204 @@
+import Flatpickr from 'flatpickr';
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch
+  from '../../globals/js/mixins/init-component-by-search';
+
+// `this.options` create-component mix-in creates prototype chain
+// so that `options` given in constructor argument wins over the one defined in static `options` property
+// 'Flatpickr' wants flat structure of object instead
+
+function flattenOptions(options) {
+  const o = {};
+  // eslint-disable-next-line guard-for-in, no-restricted-syntax
+  for (const key in options) {
+    o[key] = options[key];
+  }
+  return o;
+}
+
+class DatePicker extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * DatePicker.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an date picker.
+   */
+  constructor(element, options) {
+    super(element, options);
+    const type = this.element.getAttribute(this.options.attribType);
+    if (type === 'no-calendar') {
+      this._addInputLogic(this.element.querySelector(this.options.selectorDatePickerInput));
+    } else {
+      this.element.calendar = this._initDatePicker(type);
```

Comment:
The `.release()` code refers to `this.calendar` whereas the code here refers to `this.element.calendar`. Given `DatePicker` instance associated with a DOM element can be referred to via `DatePicker.components.get(element)`, I don't think we have to attach Flatpickr instance to DOM element. Also given this I guess some of the getters/setters for interacting with Flatpickr instance (especially ones doing direct mappings) is not necessary?